### PR TITLE
Added support for UTF-8 clipboard based on TigerVNC/TightVNC extension

### PR DESCRIPTION
**Pull request by egorksv**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -6,7 +6,7 @@
     <groupId>com.shinyhut</groupId>
     <artifactId>vernacular</artifactId>
     <packaging>jar</packaging>
-    <version>1.15-SNAPSHOT</version>
+    <version>1.15.1</version>
```
Comment (shinyhut) [inline pom.xml:9]: Undo this

File: pom.xml [hunk 2]
```diff
@@ -96,6 +96,7 @@
                         </manifest>
                     </archive>
                 </configuration>
+
```
Comment (shinyhut) [inline pom.xml:99]: Remove the blank line

File: pom.xml [hunk 3]
```diff
@@ -176,13 +192,9 @@
...
     <distributionManagement>
-        <snapshotRepository>
-            <id>ossrh</id>
-            <url>https://s01.oss.sonatype.org/content/repositories/snapshots</url>
-        </snapshotRepository>
         <repository>
-            <id>ossrh</id>
-            <url>https://s01.oss.sonatype.org/service/local/staging/deploy/maven2/</url>
+            <id>maven.nuvalence.repo</id>
+            <url>s3://testup-build-artifacts/m2</url>
```
Comment (shinyhut) [inline pom.xml:197]: Undo this

File: src/main/java/com/shinyhut/vernacular/VernacularViewer.java [hunk 4]
```diff
@@ -342,9 +343,9 @@ private String showUsernameDialog() {
     }
 
     private String showPasswordDialog() {
-        String password = "";
+        String password = "headless";
```
Comment (shinyhut) [inline src/main/java/com/shinyhut/vernacular/VernacularViewer.java:346]: Leave this empty by default

File: src/main/java/com/shinyhut/vernacular/VernacularViewer.java [hunk 5]
```diff
@@ -342,9 +343,9 @@ private String showUsernameDialog() {
...
         JPanel passwordDialog = new JPanel();
-        JPasswordField passwordField = new JPasswordField(20);
+        JPasswordField passwordField = new JPasswordField("headless",20);
```
Comment (shinyhut) [inline src/main/java/com/shinyhut/vernacular/VernacularViewer.java:348]: And this

File: src/main/java/com/shinyhut/vernacular/protocol/messages/ServerCutText.java [hunk 6]
```diff
@@ -1,29 +1,110 @@
...
+     * (@see https://github.com/TigerVNC/tigervnc/pull/834/files)
+     *
+     *
+     * @param in
+     * @param textLength
+     * @param handler
+     * @return
+     * @throws IOException
+     */
+    private static ServerCutText readExtendedClipboard(DataInputStream in, int textLength, ClipboardEventHandler handler) throws IOException {
```
Comment (shinyhut) [inline src/main/java/com/shinyhut/vernacular/protocol/messages/ServerCutText.java:64]: This should really have a unit test

File: pom.xml [hunk 7]
```diff
@@ -148,7 +149,22 @@
...
+                <configuration>
+                    <source>11</source>
+                    <target>11</target>
+                </configuration>
+            </plugin>
         </plugins>
+        <extensions>
+            <extension>
+                <groupId>com.github.seahen</groupId>
+                <artifactId>maven-s3-wagon</artifactId>
```
Comment (shinyhut) [inline pom.xml:164]: Remove this

**Review by shinyhut — CHANGES_REQUESTED** [review]
Hi @egorksv thanks for this, it looks good in principle. I've had a quick run through and I think this PR includes a lot of changes that are specific to your fork and build environment (POM changes, default passwords etc) - I think you'll need to create a branch in your fork without these changes and open a PR from there. If you get that sorted out I'll have a closer look at the logic and hopefully merge.

File: pom.xml (hunk 7 above)
Comment (egorksv) [reply pom.xml:164]: Hi Paul

We've encountered a few issues with this feature - just wanted to ensure they're all ironed out before we proceed with the pull request.

Cheers
Egor

**Pull request closed without merging**